### PR TITLE
update to new LDLib recipe viewer changes

### PR DESCRIPTION
**Pull request by screret**
## What
updates to latest LDLib to remove uses of internal recipeviewer code and switch to (mostly) API only.

## Implementation Details
added a `TagOrCycleItemStackTransfer` and `TagOrCycleFluidTransfer` in LDLib and used those for managing tags in recipe viewer display slots to allow seeing e.g. the list of valid items in REI and EMI.

## Outcome
cleaner recipe viewer widgets with some extra features like stated above

## Potential Compatibility Issues
shouldn't be, unless someone is mixining into our recipe viewer code.

### CODE REVIEW

**Comment by Arborsm** [comment]
the float item
7aacbdf825d0171bdd4f953e0ab3021d

**Comment by Arborsm** [comment]
weird group display
381b07c629839c392fe648f13ff76869

**Comment by Arborsm** [comment]
incorrect faverite behavior

https://github.com/GregTechCEu/GregTech-Modern/assets/44860537/bd62ac52-29ad-401e-b34e-1245d681fbf3

**Comment by Arborsm** [comment]
image
image
chanced things don't get a overlay like second

**Comment by Arborsm** [comment]
> weird group display
> 381b07c629839c392fe648f13ff76869
> 

it's still draggable now which makes it can't be favorite

**Comment by screret** [comment]
viewing multiblocks in EMI is still kind of.. janky? @Arborsm could you look into that after this is merged?

File: src/main/java/com/gregtechceu/gtceu/integration/GTOreByProductWidget.java
Comment (mikerooni) [inline src/main/java/com/gregtechceu/gtceu/integration/GTOreByProductWidget.java:155]: Is this by accident or necessary because of java generics having skill issues?

**Review by mikerooni — APPROVED** [review]

File: src/main/java/com/gregtechceu/gtceu/integration/GTOreByProductWidget.java
Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/integration/GTOreByProductWidget.java:155]: maps an Either<List, List> to List

**Comment by screret** [comment]
won't merge this yet because of bugs listed on the discord server. mainly to do with EMI integration.

**Review by mikerooni — APPROVED** [review]

**Pull request merged**
